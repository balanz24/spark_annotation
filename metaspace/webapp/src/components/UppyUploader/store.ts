// based on https://uppy.io/docs/stores/#Implementing-Stores

import { ref } from '@vue/composition-api'
import { State } from '@uppy/core'

type Listener = (prevState: State, nextState: State, patch: object) => void

export default () => {
  const state = ref<State>({})
  const listeners = new Set<Listener>()

  return {
    getState: () => state.value,
    setState: (patch: object) => {
      const prevState = state.value
      const nextState = { ...prevState, ...patch }

      state.value = nextState

      listeners.forEach(listener => {
        listener(prevState, nextState, patch)
      })
    },
    subscribe: (listener: Listener) => {
      listeners.add(listener)
      return () => listeners.delete(listener)
    },
  }
}
