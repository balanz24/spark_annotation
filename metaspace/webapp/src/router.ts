/* eslint-disable vue/max-len */
import Vue, { AsyncComponent } from 'vue'
import VueRouter, { RawLocation } from 'vue-router'
import AboutPage from './modules/App/AboutPage'
import DatasetsPage from './modules/Datasets/DatasetsPage.vue'
import { DialogPage, ResetPasswordPage } from './modules/Account'
import { redirectAfterSignIn } from './modules/Account/signInReturnUrl'
import { updateDBParam } from './modules/Filters/url'
import NotFoundPage from './modules/App/NotFoundPage.vue'

Vue.use(VueRouter)

const asyncPagesFreelyTyped = {
  AnnotationsPage: () => import(/* webpackPrefetch: true, webpackChunkName: "AnnotationsPage" */ './modules/Annotations/AnnotationsPage.vue'),
  DatasetSummary: () => import(/* webpackPrefetch: true, webpackChunkName: "DatasetSummary" */ './modules/Datasets/summary/DatasetSummary.vue'),
  MetadataEditPage: () => import(/* webpackPrefetch: true, webpackChunkName: "MetadataEditPage" */ './modules/MetadataEditor/MetadataEditPage.vue'),
  ImageAlignmentPage: () => import(/* webpackPrefetch: true, webpackChunkName: "ImageAlignmentPage" */ './modules/ImageAlignment/ImageAlignmentPage.vue'),
  UploadPage: () => import(/* webpackPrefetch: true, webpackChunkName: "UploadPage" */ './modules/MetadataEditor/UploadPage.vue'),
  DatasetOverviewPage: () => import(/* webpackPrefetch: true, webpackChunkName: "DatasetOverviewPage" */ './modules/Datasets/overview/DatasetOverviewPage'),
  DatasetComparisonPage: () => import(/* webpackPrefetch: true, webpackChunkName: "DatasetOverviewPage" */ './modules/Datasets/comparison/DatasetComparisonPage'),
  DatasetBrowserPage: () => import(/* webpackPrefetch: true, webpackChunkName: "DatasetOverviewPage" */ './modules/Datasets/imzml/DatasetBrowserPage'),
  DatasetEnrichmentPage: () => import(/* webpackPrefetch: true, webpackChunkName: "DatasetOverviewPage" */ './modules/Datasets/enrichment/DatasetEnrichmentPage'),
  SpottingProjectPage: () => import(/* webpackPrefetch: true, webpackChunkName: "SpottingProjectPage" */ './modules/SpottingProject/DashboardPage'),

  // These pages are relatively small as they don't have any big 3rd party dependencies, so pack them together
  DatasetTable: () => import(/* webpackPrefetch: true, webpackChunkName: "Bundle1" */ './modules/Datasets/list/DatasetTable.vue'),
  HelpPage: () => import(/* webpackPrefetch: true, webpackChunkName: "Bundle1" */ './modules/App/HelpPage.vue'),
  EditUserPage: () => import(/* webpackPrefetch: true, webpackChunkName: "Bundle1" */ './modules/UserProfile/EditUserPage.vue'),
  CreateGroupPage: () => import(/* webpackPrefetch: true, webpackChunkName: "Bundle1" */ './modules/Group/CreateGroupPage.vue'),
  ProjectsListPage: () => import(/* webpackPrefetch: true, webpackChunkName: "Bundle1" */ './modules/Project/ProjectsListPage.vue'),
  SystemHealthPage: () => import(/* webpackPrefetch: true, webpackChunkName: "Bundle1" */ './modules/Admin/SystemHealthPage.vue'),
  GroupsListPage: () => import(/* webpackPrefetch: true, webpackChunkName: "Bundle1" */ './modules/Group/GroupsListPage'),
  PrivacyPage: () => import(/* webpackPrefetch: true, webpackChunkName: "Bundle1" */ './modules/App/PrivacyPage.vue'),
  TermsPage: () => import(/* webpackPrefetch: true, webpackChunkName: "Bundle1" */ './modules/App/TermsPage.vue'),
  PublicationsPage: () => import(/* webpackPrefetch: true, webpackChunkName: "Bundle1" */ './modules/App/PublicationsPage'),

  // These pages use sanitizeHtml, which is big
  ViewGroupPage: () => import(/* webpackPrefetch: true, webpackChunkName: "Bundle2" */ './modules/Group/ViewGroupPage.vue'),
  ViewProjectPage: () => import(/* webpackPrefetch: true, webpackChunkName: "Bundle2" */ './modules/Project/ViewProjectPage.vue'),

  // Separate bundle for design docs
  DesignTOC: () => import(/* webpackChunkName: "DesignBundle" */ './design/TOCPage.vue'),
  DesignStyleGuide: () => import(/* webpackChunkName: "DesignBundle" */ './design/StyleGuidePage.vue'),
  DesignIcons: () => import(/* webpackChunkName: "DesignBundle" */ './design/IconsPage.vue'),
  DesignComponents: () => import(/* webpackChunkName: "DesignBundle" */ './design/ComponentsPage.vue'),
  DesignForms: () => import(/* webpackChunkName: "DesignBundle" */ './design/FormsPage.vue'),
}
const asyncPages = asyncPagesFreelyTyped as Record<keyof typeof asyncPagesFreelyTyped, AsyncComponent>

const convertLegacyUrls = () => {
  const { pathname, hash, search } = window.location
  if (pathname === '/' && hash && hash.startsWith('#/')) {
    history.replaceState(undefined, undefined as any, hash.slice(1))
  }
  if (pathname === '/annotations') {
    const updatedQueryString = updateDBParam(search.slice(1))
    if (updatedQueryString !== null) {
      history.replaceState(undefined, undefined as any, `${pathname}?${updatedQueryString}`)
    }
  }
}
convertLegacyUrls()

const router = new VueRouter({
  mode: 'history',
  routes: [
    { path: '/(about)?', component: AboutPage, meta: { footer: true, headerClass: 'bg-primary' } },
    { path: '/annotations', name: 'annotations', component: asyncPages.AnnotationsPage },
    {
      path: '/datasets',
      component: DatasetsPage,
      children: [
        { path: '', component: asyncPages.DatasetTable },
        { path: 'summary', component: asyncPages.DatasetSummary },
      ],
    },
    {
      path: '/datasets/:dataset_id/comparison',
      name: 'datasets-comparison',
      component: asyncPages.DatasetComparisonPage,
    },
    { path: '/datasets/edit/:dataset_id', name: 'edit-metadata', component: asyncPages.MetadataEditPage },
    { path: '/datasets/:dataset_id/add-optical-image', name: 'add-optical-image', component: asyncPages.ImageAlignmentPage },
    { path: '/dataset/:dataset_id', name: 'dataset-overview', component: asyncPages.DatasetOverviewPage },
    { path: '/dataset/:dataset_id/annotations', name: 'dataset-annotations', component: asyncPages.AnnotationsPage },
    { path: '/dataset/:dataset_id/browser', name: 'dataset-browser', component: asyncPages.DatasetBrowserPage },
    { path: '/dataset/:dataset_id/enrichment', name: 'dataset-enrichment', component: asyncPages.DatasetEnrichmentPage },
    { path: '/upload', component: asyncPages.UploadPage },
    { path: '/help', component: asyncPages.HelpPage, meta: { footer: true } },
    { path: '/user/me', component: asyncPages.EditUserPage },

    { path: '/admin/health', component: asyncPages.SystemHealthPage },

    { path: '/account/sign-in', component: DialogPage, props: { dialog: 'signIn' } },
    { path: '/account/sign-in-success', redirect: redirectAfterSignIn },
    { path: '/account/create-account', component: DialogPage, props: { dialog: 'createAccount' } },
    { path: '/account/forgot-password', component: DialogPage, props: { dialog: 'forgotPassword' } },
    { path: '/account/reset-password', component: ResetPasswordPage },

    { path: '/groups', component: asyncPages.GroupsListPage },
    { path: '/group/create', component: asyncPages.CreateGroupPage },
    { path: '/group/:groupIdOrSlug', name: 'group', component: asyncPages.ViewGroupPage },

    { path: '/project/:projectIdOrSlug', name: 'project', component: asyncPages.ViewProjectPage },
    { // Legacy URL sent in "request access" emails up until Feb 2019
      path: '/project/:projectIdOrSlug/manage',
      redirect: { path: '/project/:projectIdOrSlug', query: { tab: 'members' } } as RawLocation,
    },
    { path: '/projects', component: asyncPages.ProjectsListPage },

    { path: '/terms', component: asyncPages.TermsPage, meta: { footer: true } },
    { path: '/privacy', component: asyncPages.PrivacyPage, meta: { footer: true } },
    { path: '/publications', component: asyncPages.PublicationsPage, meta: { footer: true } },

    { path: '/design', component: asyncPages.DesignTOC, meta: { footer: true, flex: true } },
    { path: '/design/styleguide', component: asyncPages.DesignStyleGuide, meta: { footer: true, flex: true } },
    { path: '/design/icons', component: asyncPages.DesignIcons, meta: { footer: true, flex: true } },
    { path: '/design/components', component: asyncPages.DesignComponents, meta: { footer: true, flex: true } },
    { path: '/design/forms', component: asyncPages.DesignForms, meta: { footer: true, flex: true } },

    { path: '/detectability', name: 'detectability', component: asyncPages.SpottingProjectPage },

    { path: '*', component: NotFoundPage, meta: { footer: true, flex: true } },
  ],
  scrollBehavior(to, from, savedPosition) {
    if (to.hash) {
      return { selector: to.hash, offset: { x: 0, y: 64 } } // offset header height
    }
    if (savedPosition) {
      return savedPosition
    }
    if (to.path !== from.path) {
      return { x: 0, y: 0 }
    }
  },
})

const pageLoadedAt = Date.now()
router.beforeEach((to, from, next) => {
  // If user has had the window open for a long time, it's likely they're using old code. This can cause issues
  // such as incorrect API usage, or causing error reports for bugs that have already been fixed.
  // Force the browser to reload the page, preferring to do so during page transitions to minimize impact.
  const daysSincePageLoad = (Date.now() - pageLoadedAt) / 1000 / 60 / 60 / 24
  if (to.path !== from.path && daysSincePageLoad > 1.75 || daysSincePageLoad > 7) {
    window.location.assign(to.fullPath)
    next(false)
  } else {
    next()
  }
})

const { href } = router.resolve({ name: 'project', params: { projectIdOrSlug: 'REMOVE' } }, undefined, true)
export const PROJECT_URL_PREFIX = location.origin + href.replace('REMOVE', '')

export default router
