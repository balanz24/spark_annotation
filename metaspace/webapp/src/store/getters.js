import { mzFilterPrecision } from '../lib/util';
import { decodeParams, decodeSettings, getLevel } from '../modules/Filters';
import config from '../lib/config';

/** For filters where empty string is a valid client-side value that must be converted to empty string for the API */
const noneToEmptyString = s => s === 'none' ? '' : s;

export default {
  filterLevel(state) {
    return getLevel(state.route.path);
  },

  filter(state) {
    return decodeParams(state.route, state.filterLists);
  },

  currentUser(state) {
    return state.currentUser;
  },

  settings(state) {
    return decodeSettings(state.route);
  },

  ftsQuery(state, getters) {
    return getters.filter.simpleQuery;
  },

  gqlAnnotationFilter(state, getters) {
    const filter = getters.filter;
    const colocalizationAlgo = getters.settings.annotationView.colocalizationAlgo;

    const f = {
      compoundQuery: filter.compoundName,
      termId: filter.term ? parseInt(filter.term, 10) : undefined,
      chemMod: noneToEmptyString(filter.chemMod),
      neutralLoss: noneToEmptyString(filter.neutralLoss),
      adduct: filter.adduct,
      annotationId: filter.annotationIds ? filter.annotationIds.join('|') : undefined,
      fdrLevel: filter.fdrLevel,
      pValue: filter.pValue,
      colocalizedWith: filter.colocalizedWith,
      // Only include colocalizationAlgo if there is another filter that uses it. Otherwise the annotations list
      // refreshes unnecessarily when changing algorithm.
      colocalizationAlgo: filter.colocalizedWith || filter.colocalizationSamples ? colocalizationAlgo : null,
      colocalizationSamples: filter.colocalizationSamples,
      offSample: filter.offSample == null ? undefined : !!filter.offSample,
    };

    if (!config.features.neutral_losses && !config.features.advanced_ds_config) {
      f.hasNeutralLoss = false;
    }

    if (!config.features.chem_mods && !config.features.advanced_ds_config) {
      f.hasChemMod = false;
    }

    if (!config.features.all_adducts) {
      f.hasHiddenAdduct = false;
    }

    if (filter.minMSM)
      f.msmScoreFilter = { min: filter.minMSM, max: 1.0 };

    if (filter.mz) {
      const mz = parseFloat(filter.mz),
        deltamz = parseFloat(mzFilterPrecision(filter.mz));
      f.mzFilter = {
        min: (mz - deltamz),
        max: (mz + deltamz)
      };
    }

    f.databaseId = filter.database

    return f;
  },

  gqlDatasetFilter(state, getters) {
    const filter = getters.filter;
    const { group, project, submitter, datasetIds, polarity,
      organism, organismPart, condition, growthConditions,
      ionisationSource, analyzerType, maldiMatrix, metadataType,
      compoundName, datasetOwner, opticalImage } = filter;
    const level = getters.filterLevel;
    const isLogged = state.currentUser && state.currentUser.id
    const hasAnnotationMatching = level === 'dataset' && compoundName ? { compoundQuery: compoundName } : undefined;
    return {
      group: datasetOwner && datasetOwner !== 'my-datasets' && isLogged ? datasetOwner : group,
      project: project,
      submitter: datasetOwner === 'my-datasets' && isLogged ? state.currentUser.id : submitter,

      // temporary workaround because of array-related bugs in apollo-client
      ids: datasetIds ? datasetIds.join("|") : null,

      organism,
      opticalImage,
      organismPart,
      condition,
      growthConditions,
      ionisationSource,
      maldiMatrix,
      analyzerType,
      polarity: polarity ? polarity.toUpperCase() : null,
      metadataType,
      hasAnnotationMatching,
    }
  },

  gqlColocalizationFilter(state, getters) {
    const { datasetIds, colocalizedWith, database, fdrLevel } = getters.filter;
    const colocalizationAlgo = getters.settings.annotationView.colocalizationAlgo;
    if (datasetIds && !datasetIds.includes('|') && colocalizedWith != null && database != null && fdrLevel != null) {
      return { colocalizedWith, colocalizationAlgo, databaseId: database, fdrLevel };
    } else {
      return null;
    }
  },
}
