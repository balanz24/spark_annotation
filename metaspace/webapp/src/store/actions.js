import apolloClient from '../api/graphqlClient';
import {fetchOptionListsQuery} from '../api/metadata';
import {decodeParams} from '../modules/Filters';
import {computed} from "@vue/composition-api";
import store from "../store/index";
export default {

  async initFilterLists(context) {
    if(context.state.filterListsLoading || context.state.filterLists != null)
      return;

    context.commit('setFilterListsLoading');

    const response = await apolloClient.query({
      query: fetchOptionListsQuery,
      fetchPolicy: 'cache-first',
    });

    // set annotationIds default according to values passed from snapshot to the store state
    context.commit('setFilterLists', {...response.data,
      annotationIds: computed(() => store.state.snapshotAnnotationIds)} );

    // Refresh the current filter so that computed defaults that depend on `filterLists` are applied
    const filter = decodeParams(context.state.route, context.state.filterLists)
    context.commit('updateFilter', filter)
  },

};
