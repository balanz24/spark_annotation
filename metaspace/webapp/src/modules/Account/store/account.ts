import { Module } from 'vuex'
import { DialogType } from '../dialogs'
import router from '../../../router'
import { RawLocation } from 'vue-router/types/router'

export interface AccountState {
  dialog: DialogType | null;
  dialogCloseRedirect: RawLocation | null;
  loginSuccessRedirect: RawLocation | null;
}

interface ShowDialogOptions {
  dialog: DialogType;
  dialogCloseRedirect?: RawLocation;
  loginSuccessRedirect?: RawLocation;
}

interface HideDialogOptions {
  dialog: DialogType;
  isLoginSuccess?: boolean;
}

const account: Module<AccountState, any> = {
  namespaced: true,
  state: {
    dialog: null,
    dialogCloseRedirect: null,
    loginSuccessRedirect: null,
  },
  mutations: {
    showDialog(state, payload: DialogType | ShowDialogOptions) {
      const options: ShowDialogOptions = typeof payload === 'string' ? { dialog: payload } : payload

      state.dialog = options.dialog
      if (options.dialogCloseRedirect != null) {
        state.dialogCloseRedirect = options.dialogCloseRedirect
      }
      if (options.loginSuccessRedirect != null) {
        state.loginSuccessRedirect = options.loginSuccessRedirect
      }
    },
    hideDialog(state, payload: DialogType | HideDialogOptions) {
      const options: HideDialogOptions = typeof payload === 'string' ? { dialog: payload } : payload

      if (state.dialog === options.dialog) {
        state.dialog = null
        if (options.isLoginSuccess && state.loginSuccessRedirect != null) {
          router.push(state.loginSuccessRedirect)
        } else if (state.dialogCloseRedirect != null) {
          router.push(state.dialogCloseRedirect)
        }

        state.dialogCloseRedirect = null
        state.loginSuccessRedirect = null
      }
    },
  },
}

export default account
