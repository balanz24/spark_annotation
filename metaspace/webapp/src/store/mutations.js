import { isEqual, omit, without } from 'lodash-es';
import router from '../router';
import compare from '../lib/compare';
import store from "../store/index";
import {
  decodeParams,
  encodeParams,
  encodeSections,
  encodeSortOrder,
  FILTER_SPECIFICATIONS,
  getFilterInitialValue,
  stripFilteringParams,
} from '../modules/Filters';
import { DEFAULT_ANNOTATION_VIEW_SECTIONS, DEFAULT_COLORMAP, DEFAULT_TABLE_ORDER } from '../modules/Filters/url';
import { DEFAULT_SCALE_TYPE } from '../lib/constants';
import Vue from 'vue'

function updatedLocation(state, filter) {
  let query = encodeParams(filter, state.route.path, state.filterLists);

  state.lastUsedFilters[state.route.path] = {
    filter,
    query,
    order: state.orderedActiveFilters
  };

  return {
    query: Object.assign(query, stripFilteringParams(state.route.query))
  };
}

function replaceURL(state, filter) {
  router.replace(updatedLocation(state, filter));
}

function pushURL(state, filter) {
  router.push(updatedLocation(state, filter));
}

function sortFilterKeys(keys) {
  keys.sort((a, b) => compare(FILTER_SPECIFICATIONS[a].sortOrder || 100, FILTER_SPECIFICATIONS[b].sortOrder || 100));
}

function updateFilter(state, filter, routerAction = null) {
  let active = [];

  // drop unset filters
  for (var i = 0; i < state.orderedActiveFilters.length; i++) {
    let key = state.orderedActiveFilters[i];
    if (filter[key] !== undefined)
      active.push(key);
  }

  // append newly added filters to the end
  for (var key in filter)
    if (filter[key] !== undefined &&
      active.indexOf(key) == -1)
      active.push(key);

  // sort
  sortFilterKeys(active);

  const changedFilterSet = !isEqual(state.orderedActiveFilters, active);

  state.orderedActiveFilters = active;
  if (routerAction === 'push' || (routerAction == null && changedFilterSet))
    pushURL(state, filter);
  else
    replaceURL(state, filter);
}

export default {
  updateFilter(state, filter) {
    // TODO: Stop using updateFilter for adding/removing filters, as it makes reacting to added/removed filters
    // much more difficult
    updateFilter(state, filter);
  },

  updateCurrentUser(state, user) {
    state.currentUser = user
  },

  replaceFilter(state, filter) {
    updateFilter(state, filter, 'replace');
  },

  addFilter(state, name) {
    const oldFilter = decodeParams(state.route, state.filterLists);
    const filtersToAdd = [name];
    const filtersToRemove = [];

    // Check for required additional filters & conflicting filters
    (FILTER_SPECIFICATIONS[name].dependsOnFilters || []).forEach(key => {
      if (oldFilter[key] === undefined) {
        filtersToAdd.push(key);
      }
    });
    (FILTER_SPECIFICATIONS[name].conflictsWithFilters || []).forEach(key => {
      if (oldFilter[key] !== undefined) {
        filtersToRemove.push(key);
      }
    });

    const newFilter = { ...oldFilter };
    filtersToAdd.forEach(key => {
      newFilter[key] = getFilterInitialValue(key, state.filterLists);
    });
    filtersToRemove.forEach(key => {
      newFilter[key] = undefined;
    });
    const newActive = without([...state.orderedActiveFilters, ...filtersToAdd], ...filtersToRemove);
    sortFilterKeys(newActive);

    state.orderedActiveFilters = newActive;
    pushURL(state, newFilter);
  },

  removeFilter(state, name) {
    const oldFilter = decodeParams(state.route, state.filterLists);
    const filtersToRemove = [name];

    // Check for dependent filters that should also be removed
    Object.keys(oldFilter).forEach(key => {
      const { dependsOnFilters } = FILTER_SPECIFICATIONS[key];
      if (dependsOnFilters != null && dependsOnFilters.includes(name)) {
        filtersToRemove.push(key);
      }
    });

    const newFilter = omit(oldFilter, filtersToRemove);
    state.orderedActiveFilters = without(state.orderedActiveFilters, ...filtersToRemove);
    pushURL(state, newFilter);
  },

  updateFilterOnNavigate(state, to) {
    const newActiveFilters = Object.keys(decodeParams(to, state.filterLists));

    const removed = without(state.orderedActiveFilters, ...newActiveFilters);
    const added = without(newActiveFilters, ...state.orderedActiveFilters);

    state.orderedActiveFilters = without(state.orderedActiveFilters, ...removed);
    state.orderedActiveFilters.push(...added);
    sortFilterKeys(state.orderedActiveFilters);
  },

  setFilterListsLoading(state) {
    state.filterListsLoading = true;
  },

  setFilterLists(state, filterLists) {
    state.filterLists = filterLists;
    state.filterListsLoading = false;
  },

  setAnnotation(state, annotation) {
    state.annotation = annotation;
  },

  setNormalizationMatrix(state, normalizationMatrix) {
    state.normalization = normalizationMatrix;
  },

  resetChannels(state) {
    state.channels = [];
  },

  addChannel(state, {id, annotations, settings}) {
    state.channels.push({id, annotations, settings});
  },

  restoreChannels(state, channels) {
    state.channels = channels;
  },

  removeChannel(state, {index}) {
    const aux = state.channels.slice(0)
    aux.splice(index, 1)
    state.channels = aux
  },

  updateChannel(state, {index, id, annotations, settings}) {
    const aux = state.channels.slice(0)
    aux[index] = {id, annotations, settings}
    state.channels = aux
  },

  setViewerMode(state, mode = 'SINGLE') {
    state.mode = mode;
  },


  setRoiInfo(state, {key, roi}) {
    Vue.set(state.roiInfo, key, roi)
  },

  toggleRoiVisibility(state, visible) {
    Vue.set(state.roiInfo, 'visible', visible)
  },

  resetRoiInfo(state) {
    state.roiInfo = {visible: false};
  },

  setSnapshotAnnotationIds(state, annotation) {
    state.snapshotAnnotationIds = annotation;
  },

  updateAnnotationTableStatus(state, isLoading) {
    state.tableIsLoading = isLoading;
  },

  startTour(state, tourDescription) {
    state.currentTour = tourDescription;
  },

  endTour(state) {
    state.currentTour = null;
  },

  updateAnnotationViewSections(state, activeSections) {
    const sections = encodeSections(activeSections);
    const defaultSections = encodeSections(DEFAULT_ANNOTATION_VIEW_SECTIONS);
    router.replace({
      query: sections !== defaultSections
        ? { ...state.route.query, sections }
        : omit(state.route.query, 'sections'),
    });
  },

  setColormap(state, cmap) {
    router.replace({
      query: cmap !== DEFAULT_COLORMAP
        ? { ...state.route.query, cmap }
        : omit(state.route.query, 'cmap'),
    });
  },

  setScaleType(state, scaleType) {
    router.replace({
      query: scaleType !== DEFAULT_SCALE_TYPE
        ? { ...state.route.query, scale: scaleType }
        : omit(state.route.query, 'scale'),
    });
  },

  setLockTemplate(state, lockTemplate) {
    router.replace({
      query: lockTemplate
        ? { ...state.route.query, lock: lockTemplate }
        : omit(state.route.query, 'lock'),
    });
  },

  setNormalization(state, normalization) {
    router.replace({
      query: !normalization ? omit(state.route.query, 'norm')
        : {...state.route.query, norm: normalization }
    });
  },

  setCurrentPage(state, page) {
    router.replace({
      query: page !== 1
        ? { ...state.route.query, page: String(page) }
        : omit(state.route.query, 'page'),
    });
  },

  setRow(state, row) {
    router.replace({
      query: row !== 1
        ? { ...state.route.query, row: String(row) }
        : omit(state.route.query, 'row'),
    });
  },

  setSortOrder(state, sortOrder) {
    const sort = encodeSortOrder(sortOrder);
    const defaultSort = encodeSortOrder(DEFAULT_TABLE_ORDER);
    router.replace({
      query: sort !== defaultSort
        ? { ...state.route.query, sort }
        : omit(state.route.query, 'sort'),
    });
  },

  setColocalizationAlgo(state, colocalizationAlgo) {
    router.replace({
      query: colocalizationAlgo != null
        ? { ...state.route.query, alg: colocalizationAlgo }
        : omit(state.route.query, 'alg'),
    });
  },

  setDatasetTab(state, tab) {
    const path = {
      'list': '/datasets',
      'summary': '/datasets/summary'
    }[tab.toLowerCase()];
    router.replace({
      path,
      query: state.route.query,
    });
  }
};
