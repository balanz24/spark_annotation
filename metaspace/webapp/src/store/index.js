import Vue from 'vue';
import Vuex from 'vuex';
Vue.use(Vuex);

import getters from './getters.js';
import mutations from './mutations.js';
import actions from './actions.js';
import {accountModule} from '../modules/Account';

const store = new Vuex.Store({
  state: {
    // names of currently shown filters
    orderedActiveFilters: [],

    filterLists: null,
    filterListsLoading: false,

    // currently selected annotation
    annotation: undefined,

    // currently selected normalization matrix
    normalization: undefined,

    // roi settings
    roiInfo: {visible: false},

    // is annotation table loading?
    tableIsLoading: true,

    lastUsedFilters: {},

    currentTour: null,

    currentUser: {},

    // ion image global viewer settings
    channels: [],
    mode: 'SINGLE'
  },

  getters,
  mutations,
  actions,
  modules: {
    account: accountModule
  }
})

export default store;
