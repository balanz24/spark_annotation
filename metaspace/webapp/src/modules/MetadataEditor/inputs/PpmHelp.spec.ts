import { mount } from '@vue/test-utils'
import PpmHelp from './PpmHelp.vue'
import router from '../../../router'
import store from '../../../store/index'
import Vue from 'vue'

describe('PpmHelp', () => {
  it('should match snapshot', async() => {
    const wrapper = mount(PpmHelp, { store, router })
    await Vue.nextTick()

    expect(wrapper).toMatchSnapshot()
  })
})
