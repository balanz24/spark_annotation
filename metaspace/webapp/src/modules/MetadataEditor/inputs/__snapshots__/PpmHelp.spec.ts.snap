// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PpmHelp should match snapshot 1`] = `
<div style="max-width: 500px;">
  <p>
    For mass resolving power below 70K, we recommend using higher m/z tolerance of 5-10 ppm. Please keep in mind that
    increasing tolerance leads to increased ambiguity, often lower numbers of annotations at fixed FDR, and increased
    numbers of isobaric annotations.
  </p>
</div>
`;
